src/agora/dodemul.cc

```diff
@@ -306,29 +306,65 @@ EventData DoDemul::Launch(size_t tag) {
       equal_ptr += cfg_->UeAntNum() * k_num_double_in_sim_d256 * 2;
     }
     equal_t_ptr = (float*)(equaled_buffer_temp_transposed_);
-    int8_t* demod_ptr = demod_buffers_[frame_slot][symbol_idx_ul][i] +
-                        (cfg_->ModOrderBits() * base_sc_id);
+    int8_t* demod_ptr = demod_buffers_[frame_slot][symbol_idx_ul][ue_id] +
+                        (cfg_->ModOrderBits(Direction::kUplink) * base_sc_id);
 
-    switch (cfg_->ModOrderBits()) {
+    switch (cfg_->ModOrderBits(Direction::kUplink)) {
```

Comment:
We need to add some comments here about what the difference between hard demodulation and soft demodulation is because it is not very intuitive.